Amethyst/Managers/WindowTransitionCoordinator.swift

```diff
@@ -52,14 +52,17 @@ class WindowTransitionCoordinator<Target: WindowTransitionTarget> {
 
         if windows.count == 1 {
             return
-        } else if windows.count == 2 || target?.currentLayout()?.layoutKey == TwoPaneLayout<Window>.layoutKey {
-            // Swap the two visible windows, keep focus on the main window if it already was there
-            target?.executeTransition(.switchWindows(focusedWindow, windows[1 - focusedIndex]))
-            if focusedWindow == windows[0] {
-                windows[1 - focusedIndex].focus()
-            }
-        } else {
-            // Swap focused window with main window
+        }
+
+        if focusedIndex == 0 && target?.currentLayout()?.layoutKey == TwoPaneLayout<Window>.layoutKey {
```

Comment:
We check for `windows.count == 1`, but we could theoretically have no windows, so I think we need to make sure that is the case before we check index `1`. Or maybe change the check on the windows count above to be `<= 1`.

Reply:
Thanks, missed that.